### PR TITLE
[stable/prometheus-blackbox-exporter] Fixes issue where containers in deployment were running as root

### PR DESCRIPTION
**Pull request by anguswilliams**
Signed-off-by: Angus Williams <anguswilliams@gmail.com>

#### What this PR does / why we need it:
Due the the `securityContext` key being duplicated in the deployment template, some of the `runAsUser` configuration was being ignored. This breaks the helm deployment in some k8s clusters which enforce that containers run as non-root.

Also updates the configmap-reload container to run as user `65534` as per the jimmidyson/configmap-reload [Dockerfile](https://github.com/jimmidyson/configmap-reload/blob/master/Dockerfile#L2).

#### Which issue this PR fixes
N/A

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by fulii** [comment]
Hello!

This chart will break icmp blackbox probing with podsecuritypolicy enabled clusters.

File: stable/prometheus-blackbox-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -51,8 +51,9 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           securityContext:
+            readOnlyRootFilesystem: true
```
Comment (cpanato) [inline stable/prometheus-blackbox-exporter/templates/deployment.yaml]: mak this configurable and set the default value

File: stable/prometheus-blackbox-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -83,6 +82,9 @@ spec:
         - name: configmap-reload
           image: "{{ .Values.configmapReload.image.repository }}:{{ .Values.configmapReload.image.tag }}"
           imagePullPolicy: "{{ .Values.configmapReload.image.pullPolicy }}"
+          securityContext:
+            runAsNonRoot: true
+            runAsUser: {{ .Values.configmapReload.runAsUser }}
```
Comment (cpanato) [inline stable/prometheus-blackbox-exporter/templates/deployment.yaml:89]: same here

File: stable/prometheus-blackbox-exporter/templates/deployment.yaml (hunk 1 above)
Comment (anguswilliams) [reply stable/prometheus-blackbox-exporter/templates/deployment.yaml]: Thanks @cpanato , PR updated

**Comment by desaintmartin** [comment]
Thanks!
/ok-to-test
/lgtm

**Pull request merged**